core-bundle/src/Command/ResizeImagesCommand.php

```diff
@@ -0,0 +1,260 @@
+        $this->storage = $storage;
+        $this->filesystem = $filesystem ?? new Filesystem();
+        $this->terminalWidth = (new Terminal())->getWidth();
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setName('contao:resize-images')
+            ->setDefinition([
+                new InputOption('time-limit', 'l', InputArgument::OPTIONAL, 'Time limit in seconds', '0'),
+                new InputOption('concurrent', 'c', InputArgument::OPTIONAL, 'Run multiple processes concurrently', '1'),
+                new InputOption('throttle', 't', InputArgument::OPTIONAL, 'Pause between resizes to limit CPU utilization, 0.1 relates to 10% CPU usage', '1'),
+            ])
+            ->setDescription('Resizes deferred images that have not been processed yet.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if (!$this->resizer instanceof DeferredResizerInterface) {
+            throw new \RuntimeException('Deferred resizer not available');
+        }
+
+        $timeLimit = (float) $input->getOption('time-limit');
+        $throttle = (float) $input->getOption('throttle');
+        $concurrent = (int) $input->getOption('concurrent');
+
+        if ($timeLimit < 0) {
+            throw new InvalidArgumentException(sprintf('Time-limit value "%s" is invalid.', $timeLimit));
+        }
+
+        if ($throttle < 0.001 || $throttle > 1) {
+            throw new InvalidArgumentException(sprintf('Throttle value "%s" is invalid.', $throttle));
+        }
+
+        if ($concurrent < 1) {
+            throw new InvalidArgumentException(sprintf('Concurrent value "%s" is invalid.', $concurrent));
+        }
+
+        if ($concurrent > 1) {
+            return $this->executeConcurrent($concurrent, $input, $output);
+        }
+
+        $startTime = microtime(true);
+
+        $maxDuration = 0;
+        $lastDuration = 0;
+
+        foreach ($this->storage->listPaths() as $path) {
+            $sleep = (1 - $throttle) / $throttle * $lastDuration;
+
+            if ($timeLimit && microtime(true) - $startTime + $maxDuration + $sleep > $timeLimit) {
+                $output->writeln("\n".'Time limit of '.$timeLimit.' seconds reached.');
+
+                return 0;
+            }
+
+            usleep((int) ($sleep * 1000000));
+
+            $maxDuration = max($maxDuration, $lastDuration = $this->resizeImage($path, $output));
+        }
+
+        $output->writeln("\n".'All images resized.');
+
+        return 0;
+    }
+
+    private function resizeImage(string $path, OutputInterface $output): float
+    {
+        $startTime = microtime(true);
+
+        if ($this->filesystem->exists($path)) {
+            return 0;
+        }
+
+        $output->write(str_pad($path, $this->terminalWidth - 14, '.').' ');
```

Comment:
@ausi Just a cosmetic issue: This does not work everytime - sometimes there's a dot missing (eg. if there's a  `ß`  in the pathname). I guess we'd need sth like `mb_str_pad` (if it existed) for full unicode support.

Reply:
How about this?

```suggestion
        $output->write(str_pad($path, $this->terminalWidth + (strlen($path) - mb_strlen($path, 'UTF-8')) - 14, '.').' ');
```

Reply:
:+1: that works. 

I thought about this:
```suggestion
        $output->write($path . implode('', array_fill(0, $this->terminalWidth - mb_strlen($path, 'UTF-8') - 14, '.')));
```
... but it trades 'hacking str_pad to work right' against 'misusing arrays/implode'. I actually like yours better.